source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1054 @@
+                            }
+                          else
+                            data_table(idx)= 0;
+
+                        }
+                    }
+                }
+            }
+
+          Functions::InterpolatedUniformGridData<dim> *velocities;
+          velocities = new Functions::InterpolatedUniformGridData<dim> (grid_extent,
+                                                                        table_intervals,
+                                                                        data_table);
+
+          auto lambda = [&](const Point<dim> &p) -> double
+          {
+            return velocities->value(p);
+          };
+
+          VectorFunctionFromScalarFunctionObject<dim> vector_function_object(
+            lambda,
+            dim-1,
+            dim);
+
+          VectorTools::interpolate_boundary_values (mesh_deformation_dof_handler,
+                                                    *boundary_ids.begin(),
+                                                    vector_function_object,
+                                                    mesh_velocity_constraints);
+
+        }
+    }
+
+    //TODO: Give better explanations of variables and cite the fastscape documentation.
+    template <int dim>
+    void FastScape<dim>::declare_parameters(ParameterHandler &prm)
+    {
+
+      prm.declare_entry ("End time",
+                         /* boost::lexical_cast<std::string>(std::numeric_limits<double>::max() /
+                                              year_in_seconds) = */ "5.69e+300",
+                         Patterns::Double (),
+                         "The end time of the simulation. The default value is a number "
+                         "so that when converted from years to seconds it is approximately "
+                         "equal to the largest number representable in floating point "
+                         "arithmetic. For all practical purposes, this equals infinity. "
+                         "Units: Years if the "
+                         "'Use years in output instead of seconds' parameter is set; "
+                         "seconds otherwise.");
+
+      prm.enter_subsection ("Mesh deformation");
+      {
+        prm.enter_subsection ("Fastscape");
+        {
+          prm.declare_entry("Number of steps", "10",
+                            Patterns::Integer(),
+                            "Number of steps per ASPECT timestep");
+          prm.declare_entry("Maximum timestep", "10e3",
+                            Patterns::Double(0),
+                            "Maximum timestep for fastscape.");
+          prm.declare_entry("Vertical exaggeration", "3",
+                            Patterns::Double(),
+                            "Vertical exaggeration for fastscape's VTK file.");
+          prm.declare_entry("Additional fastscape refinement", "0",
+                            Patterns::Integer(),
+                            "Refinement level expected at surface to determine"
```

Comment:
From the name and description it's not clear how this differs from `Surface resolution`